### DISCUSSION TITLE
How accurately ImageMagick compare if one image was saved from template and other one is from website?

### DISCUSSION THREAD

**Discussion by SangsSharma** [ImageMagick/ImageMagick#5967] [Consulting]
Hi,
I have a case where I am looking for a solution to compare different images. This would validate the differences between the design copy and the developer's implementation.

Image 1- is taken from an XD design template
Image 2- screenshot from the website

Checking if ImageMagick provides such a feature.

Thanks.

**Comment by fmw42** [comment]
Imagemagick has the magick compare feature, which compares images pixel by pixel if the same size or if one is larger than the other, it will search for the best match region and report the match score.

Is that what you are looking for? 

There are many metrics that can be used. But one is the perceptual hash metric. 

See 

https://imagemagick.org/script/compare.php
https://imagemagick.org/Usage/compare/
http://www.fmwconcepts.com/misc_tests/perceptual_hash_test_results_510/index.html

**Reply by SangsSharma** [reply]
Thanks for the details. However, in my case, the purpose of the comparison is to find the deviations between the design template and the actual implementation by dev. This can be anything, such as a text mismatch of the size of the web element.

**Comment by fmw42** [comment]
One mode of the output is to show you on a pixel-by-pixel basis where there are differences in the two images. It will show an image. See the examples in the links.

Alternately, you can just do a -compose difference -composite between the two images to make an image showing the differences on a pixel-by-pixel difference.

Perhaps you would like to provide a pair of example images.

**Reply by SangsSharma** [reply]
Thanks, 
I will go through the links and check if it meets the requirements.
